packages/graphql/src/plugins/nexusDeferIfNotLoadedPlugin.ts

```diff
@@ -0,0 +1,117 @@
+import { plugin } from 'nexus'
+import debugLib from 'debug'
+import { getNamedType, isNonNullType } from 'graphql'
+import type { DataContext } from '@packages/data-context'
+
+const NO_RESULT = {}
+// 2ms should be enough time to resolve from the local cache of the
+// cloudUrqlClient in CloudDataSource
+const RACE_MAX_EXECUTION_MS = 2
+const IS_DEVELOPMENT = process.env.CYPRESS_INTERNAL_ENV !== 'production'
+const debug = debugLib('cypress:graphql:nexusDeferIfNotLoadedPlugin')
+
+/**
+ * This plugin taps into each of the requests and checks for the existence
+ * of a "Cloud" prefixed type. When we see these, we know that we're dealing
+ * with a remote API. We can also specify `deferIfNotLoaded: true` on the Nexus definition
+ * to indicate that this is a remote request, such as resolving the "versions" field
+ */
+export const nexusDeferIfNotLoadedPlugin = plugin({
+  name: 'nexusDeferIfNotLoadedPlugin',
+  fieldDefTypes: 'deferIfNotLoaded?: true',
+  onCreateFieldResolver (def) {
+    const { name: parentTypeName } = def.parentTypeConfig
+
+    // Don't ever need to do this on Subscription / Mutation fields,
+    // or on the Cloud types themselves, since these don't actually need to resolve themselves,
+    // they're resolved from the remote request
+    if (parentTypeName === 'Mutation' || parentTypeName === 'Subscription' || parentTypeName.startsWith('Cloud')) {
```

Comment:
switching behavior off of name prefixes feels dirty. is there a better way to accomplish this?

Reply:
Types in the `cloud.graphql` schema are consistently named with `Cloud*` so I don't think it's terrible, we could check against things defined in that schema, but then we'd need to have a separate list of things that can be used in both, like `PageInfo` & `Node`. I think this is probably simplest for now.

Reply:
I was thinking something like adding a `noDefer` field... problem is, someone will add something without the `Cloud` prefix eventually, and this line will stop doing what it's supposed to do. Moving this to the definition of the type itself would make it a lot clearer to future devs what's happening.

Reply:
> someone will add something without the Cloud prefix eventually

That’s a process issue we can/will address on the cloud side. We already have pretty good static checks in place

Reply:
Unless you’re actually saying someone will add it in the local typedefs. I can also add something that guards on that

Reply:
Updated - it's a bit more explicit now